File: Algorithm/subjects/1-100/11. 盛最多水的容器/main.java
class Solution {
    public int maxArea(int[] height) {
        int length = height.length;
        int left = 0, right = length - 1;
        int sum = -1;
        while (left < right) {
            int temp = (right - left) * (Math.min(height[left],height[right]));
            sum = Math.max(sum, temp);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return sum;
    }
}

File: Algorithm/subjects/1-100/11. 盛最多水的容器/main.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int left = 0, right = n - 1;
        int maxVal = 0;
        while (left < right) {
            int area = min(height[left], height[right]) * (right - left);
            maxVal = max(area, maxVal);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxVal;
    }
};